src/solvers/smt2_incremental/smt_terms.cpp

```diff
@@ -67,13 +70,27 @@ smt_identifier_termt::smt_identifier_termt(irep_idt identifier, smt_sortt sort)
     is_valid_smt_identifier(identifier),
     R"(Identifiers may not contain | characters.)");
   set(ID_identifier, identifier);
+  for(auto &index : indices)
+  {
+    get_sub().push_back(
+      smt_indext::storert<smt_identifier_termt>::upcast(std::move(index)));
```

Comment:
Apologies, not so familiar with the mechanics of `std::move`.

There's already an `std::move` inside of `upcast`. Do we need this one here as well?

Reply:
You can (mostly) just think of `std::move` as saying "I'm giving you the ownership/responsibility to clean up this thing"  - effectively pass a reference/scope rather than a copy - so just because `upcast` does a `std::move` doesn't mean you don't _also_ want to do it outside of `upcast`. `upcast` can't _claim_/pass on ownership of something it's not already been _given_ ownership of.